### DISCUSSION TITLE
How to inject children to the Head in _document.js?

### DISCUSSION THREAD

**Discussion by dhhanush** [vercel/next.js#22263] [Help]
How to inject a script tag dynamically on load as a child to the Head tag in the _document.js 

I want to import an SDK on load.

I know it can be done via a HOC, a custom Page wrapper component or so. But I want to do it in _document.js because I want it to persist and don't want to initialize it for every page request.

Thank You

**Comment by michrome** [comment]
Below is the default `/pages/_document.js`. According to [the docs](https://nextjs.org/docs/advanced-features/custom-document):

> The `<Head />` component used here is not the same one from `next/head`. **The `<Head />` component used here should only be used for any `<head>` code that is common for all pages.**

So I  _think_ you can replace `<Head />` with `<Head><!--this is common for all pages --></Head>`

```js
import Document, { Html, Head, Main, NextScript } from 'next/document'

class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const initialProps = await Document.getInitialProps(ctx)
    return { ...initialProps }
  }

  render() {
    return (
      <Html>
        <Head />
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

export default MyDocument
```

**Reply by dhhanush** [reply]
Thanks for the reply Mike, I had tried that and it worked, but I wasn't sure, that's the recommended way of doing it.
